### PR TITLE
[LC-1000] Distinguish Consensus essential data and Create backup db.

### PR DESCRIPTION
**Pull request by windies21**
- Distinguish Consensus essential data.
- The node must be able to request and deliver consensus essential data.
- Organize additional items for background sync and fast start
- Separate essential data and check normal operation of existing node using separated data
- Requesting essential data with cli to create backup db (loopchain_tools)

### CODE REVIEW

File: loopchain/launcher.py [hunk 1]
```diff
@@ -198,11 +198,11 @@ def start_as_tool(args, quick_command):
     print_prologue()
 
     try:
-        from _tools.loopchain_private_tools.demotool import DemoTool
+        from loopchain_tools.tools.basictool import BasicTool
```
Comment (hyerim-kim) [inline loopchain/launcher.py:201]: Did the new path to be applied to the tools project before here?

File: loopchain/utils/__init__.py [hunk 2]
```diff
@@ -142,14 +142,10 @@ def get_stub_to_server(target, stub_class, ssl_auth_type: conf.SSLAuthType = con
 
 
 def normalize_request_url(url_input, version=None, channel=None):
```
Comment (hyerim-kim) [inline loopchain/utils/__init__.py:144]: How about this ?
```suggestion
def normalize_request_url(url_input, version=None, channel=None):

    if url_input and 'http://' in url_input:
        url_input = url_input.split("http://")[1]

    use_https = 'https://' in url_input
    if not url_input:  # ex) '' => http://localhost:9000/api/v3
        url = generate_url_from_params(version=version, channel=channel)
    elif use_https and url_input.count(':') == 1:  # ex) https://testwallet.icon.foundation
        url = generate_url_from_params(dns=url_input.split("https://")[1],
                                       version=version,
                                       use_https=True,
                                       channel=channel)
    elif use_https and url_input.count(':') == 2:  # ex) https://127.0.0.1:9000
        ip_port = url_input.split("https://")[1]
        url = generate_url_from_params(ip=ip_port.split(':')[0],
                                       port=ip_port.split(':')[1],
                                       version=version,
                                       use_https=True,
                                       channel=channel)
    elif url_input.isdigit():  # ex) 9000
        url = generate_url_from_params(port=url_input, version=version, channel=channel)
    elif ':' in url_input and url_input.split(':')[1].isdigit():  # ex) 127.0.0.1:9000, {peer_name}:9000
        url = generate_url_from_params(ip=url_input.split(":")[0],
                                       port=url_input.split(":")[1],
                                       version=version,
                                       channel=channel)
    elif url_input.count('.') == 3 and url_input.replace(".", "").isdigit():  # ex) 127.0.0.1
        url = generate_url_from_params(ip=url_input, version=version, channel=channel)
    else:  # ex) testwallet.icon.foundation => https://testwallet.icon.foundation/api/v3
        url = generate_url_from_params(dns=url_input, version=version, use_https=use_https, channel=channel)

    return url
```

File: loopchain/blockchain/blockchain.py [hunk 3]
```diff
@@ -296,6 +298,26 @@ def __remove_block_up_to_target(self, target_block: 'Block'):
...
+        return synced_block_height
+
+    def get_block(self, block_hash, block_height):
+        try:
+            block_dump = self._blockchain_store.get(block_hash.encode(encoding='UTF-8'))
+            block_version = self.__block_versioner.get_version(block_height)
+            block_serializer = BlockSerializer.new(block_version, self.__tx_versioner)
+            block = block_serializer.deserialize(json.loads(block_dump))
+        except KeyError:
+            block = self.__block_manager.block_request_by_citizen(block_height)
```
Comment (hyerim-kim) [inline loopchain/blockchain/blockchain.py:316]: How about editing the method name from `by` to `as` if `block_request_by_citizen` means that this node requests a block as a citizen? If or not, skip this comment, please.

File: loopchain/launcher.py (hunk 1 above)
Comment (windies21) [reply loopchain/launcher.py:201]: Yes,  There is loopchain_tools public(github) version. see. https://github.com/icon-project/loopchain_tools

File: loopchain/blockchain/backup_manager.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    def write_preps(roothash: 'Hash32', backup_store, blockchain: 'BlockChain'):
+        preps = blockchain.find_preps_by_roothash(roothash)
+        if preps:
+            blockchain.write_preps(roothash, preps, backup_store)
+
+    async def make_backup(self, blockchain, block_height):
+        if type(block_height) == str:
+            block_height = int(block_height)
+
+        __db_name = f"backup_db_{block_height}"
```
[the remark this answers is not in the document]
Comment (windies21) [reply loopchain/blockchain/backup_manager.py:38]: I'm fully agree your opinion.

File: loopchain/blockchain/backup_manager.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                tx_hash_value
+            )
+
+    @staticmethod
+    def write_preps(roothash: 'Hash32', backup_store, blockchain: 'BlockChain'):
+        preps = blockchain.find_preps_by_roothash(roothash)
+        if preps:
+            blockchain.write_preps(roothash, preps, backup_store)
+
+    async def make_backup(self, blockchain, block_height):
```
[the remark this answers is not in the document]
Comment (windies21) [reply loopchain/blockchain/backup_manager.py:34]: I think these data that is not dependent and seems to be okay to overwrite. So, batch is not required in here.

**Pull request merged**